src/hostapi/dsound/pa_win_ds.c

```diff
@@ -203,9 +204,8 @@ static signed long GetStreamReadAvailable( PaStream* stream );
 static signed long GetStreamWriteAvailable( PaStream* stream );
 
 
-/* FIXME: should convert hr to a string */
 #define PA_DS_SET_LAST_DIRECTSOUND_ERROR( hr ) \
-    PaUtil_SetLastHostErrorInfo( paDirectSound, hr, "DirectSound error" )
+    PaWinUtil_ConvertSystemError( paDirectSound, hr )
```

Comment:
According to stackoverflow (https://stackoverflow.com/questions/13543195/how-can-i-retrieve-error-strings-for-directx-9 and https://walbourn.github.io/wheres-dxerr-lib/) FormatMessage seems to only work with DirectSound on Windows 8 and higher, before that deprecated DXGetErrorString() and DXGetErrorDescription() seem the correct way.

Reply:
Windows 7 is still a valid target so this is reasonable feedback. Probably best option is to use the DirectSound deprecated APIs for pa_win_ds.c, and possibly provide a conditional compliation to use the new `PaWinUtil_ConvertSystemError` function if and only if targeting Windows 8 and higher.

Reply:
Unfortunately I do not have a Windows 7 machine to test DXGetErrorString so let's leave it for a different PR!
I've provided conditional compilation so that for Windows 7 and earlier old code is used.